src/states/node.rs

```diff
@@ -918,6 +922,40 @@ impl Node {
         }
     }
 
+    // Tell all neighbouring sections that our member list changed.
+    // Currently we only send this when nodes join and it's only used to add missing members.
+    fn send_section_update(&mut self) {
+        trace!("{:?} Sending section update", self);
+        let names = self.peer_mgr.routing_table().our_names();
+        // TODO: why does the method return a HashSet? A Vec is fine for us
```

Comment:
Yes, I wish there was a way to return an iterator without (a) exposing implementation details (by reusing whatever type the iterator would be internally) or (b) bothering to create a custom iterator. But for anything internal-only (which `PeerManager` appears to be) it's probably not a big issue to expose the implementation details.